### PR TITLE
mxfp4 quantize kernel

### PR DESCRIPTION
**Pull request by yilinzhao01**
## Motivation

Add single-kernel MXFP4 quantization (`quant_mxfp4_even_round`) that fused quantize mxfp4+ weight/scale shuffle in one HIP kernel. Using even_round scale algorithm which is aligned with the existing Triton kernel _mxfp4_quant_op in aiter/ops/triton/_triton_kernels/quant/quant.py.

## Comparison of kernel performance
| Shape | Scenario | quant_mxfp4_even_round_hip(us) | dynamic_per_group_scaled_quant_fp4+torch shuffle (us) |
|---|---|---|---|
| 4096×2560 | E8M0 shuffle | 8.0 | 32.6 |
| 4096×2560 | A16W4 shuffle | 8.0 | 54.4 |
| 4096×1280 | E8M0 shuffle | 5.6 | 30.9 |
| 4096×1280 | A16W4 shuffle | 4.7 | 51.1 |
| 4096×256 | E8M0 shuffle | 3.2 | 27.6 |
| 4096×256 | A16W4 shuffle | 2.8 | 48.9 |
 
## Precision verified
Sglang gsm8k test with Qwen3.5-397B
quant_mxfp4_even_round_kernel:
image

Triton kernel _mxfp4_quant_op used in sglang:
image

## Testcase
- no_shuffle: 7 shapes × bf16/fp16, vs Python reference (even_round scale + e2m1 quant)
- E8M0 shuffle: 8 shapes × bf16/fp16, vs `fp4_scale_shuffle_id` index + `shuffle_weight(layout=(16,16))`
- A16W4 shuffle: 5 shapes × bf16/fp16 × gate_up, vs `shuffle_scale_a16w4` + `shuffle_weight_a16w4`
- Edge cases: all-zero, large/tiny values, E2M1 boundary midpoints

### CODE REVIEW

File: aiter/ops/quant.py [hunk 1]
```diff
@@ -684,6 +684,62 @@ def fused_dynamic_mxfp4_quant_moe_sort_hip(
...
+    out_scale: torch.Tensor,
+    group_size: int = 32,
+    e8m0_shuffle: bool = False,
+    a16w4_shuffle: bool = False,
+    gate_up: bool = False,
+    shuffle_weight: bool = False,
+) -> None: ...
+
+
+def quant_mxfp4_even_round_hip(
```
Comment (yzhou103) [inline aiter/ops/quant.py]: It may be better to add a roundmode parameter not in the kernel name here, so we can support other rounding mode in the future.
example:
enum class MxFp4RoundMode : int { Even = 0, Stochastic = 1, Rna = 2, Trunc = 3 };

template <typename ftype, MxFp4RoundMode rmode, bool ss, bool a16, bool sw>
__global__ void quant_mxfp4_kernel(...);

Comment (yilinzhao01) [reply aiter/ops/quant.py]: done. Renamed quant_mxfp4_even_round to quant_mxfp4 with a round_mode parameter (default Even). The kernel template is parameterized on MxFp4RoundMode, currently only even is implemented — other modes will be easy to add when needed.

File: aiter/ops/quant.py [hunk 2]
```diff
@@ -684,6 +684,65 @@ def fused_dynamic_mxfp4_quant_moe_sort_hip(
...
+    round_mode: int = 0,
+    e8m0_shuffle: bool = False,
+    a16w4_shuffle: bool = False,
+    gate_up: bool = False,
+    shuffle_weight: bool = False,
+) -> tuple[torch.Tensor, torch.Tensor]:
+    assert x.is_contiguous() and x.dim() == 2
+    assert x.dtype in (torch.float16, torch.bfloat16)
+    rows, cols = x.shape
+    assert cols % group_size == 0
```
Comment (yzhou103) [inline aiter/ops/quant.py:720]: Could we add a docstring on `quant_mxfp4_hip` and a comment on
`MxFp4RoundMode::Even` explaining what "Even" means, which arch implements
it natively (gfx950 HW builtin = exact RNE) vs approximately (gfx942
fallback = round-half-away)? Right now `round_mode: int` has no context for
callers and the cross-arch divergence is invisible.

File: op_tests/test_quant_mxfp4.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+import pytest
+import torch
+
+from aiter.ops.quant import quant_mxfp4_hip
+from aiter.ops.shuffle import shuffle_scale_a16w4, shuffle_weight, shuffle_weight_a16w4
+
+
+def even_round_scale(max_abs: torch.Tensor) -> torch.Tensor:
+    max_abs_f32 = max_abs.to(torch.float32).clone()
```
Comment (yzhou103) [inline op_tests/test_quant_mxfp4.py:20]: please refers to other tests like test_gemm_a8w8.py

File: aiter/ops/quant.py (hunk 2 above)
Comment (yilinzhao01) [reply aiter/ops/quant.py:720]: done

File: op_tests/test_quant_mxfp4.py (hunk 3 above)
Comment (yilinzhao01) [reply op_tests/test_quant_mxfp4.py:20]: done.

**Review by yzhou103 — DISMISSED** [review]

**Review by yzhou103 — APPROVED** [review]
